code_generator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt

```diff
@@ -9,493 +9,333 @@
-
-<# if (componentDetails.IsBlittable) { #>
-                view.SetComponentData(entity, componentData);
-<# } else { #>
-                view.SetComponentObject(entity, componentData);
-<# } #>
-
-<# if (fieldDetailsList.Count > 0) { #>
-                var componentFieldsUpdated = false;
-                var gdkUpdate = new <#= componentDetails.TypeName #>.Update();
-<# foreach (var fieldDetails in fieldDetailsList) { #>
-                if (update.<#= fieldDetails.CamelCaseName #>.HasValue)
+                else if (!entityManager.HasComponent<ComponentRemoved<<#= componentDetails.TypeName #>>>(entity))
                 {
-                    componentFieldsUpdated = true;
-                    gdkUpdate.<#= fieldDetails.PascalCaseName #> = new Option<<#= fieldDetails.Type #>>(<#= UnityTypeMappings.GetNativeTypeMethod(fieldDetails.RawFieldDefinition, "update." + fieldDetails.CamelCaseName + ".Value", enumSet) #>);
+                    entityManager.AddComponentData(entity, new ComponentRemoved<<#= componentDetails.TypeName #>>());
                 }
-<# } #>
-
-                if (componentFieldsUpdated)
+                else 
                 {
-                    view.AddComponentsUpdated(entity, gdkUpdate, UpdatesPool);
+                    LogDispatcher.HandleLog(LogType.Error, new LogEvent(ReceivedDuplicateComponentRemoved)
+                        .WithField(LoggingUtils.LoggerName, LoggerName)
+                        .WithField(LoggingUtils.EntityId, op.EntityId.Id)
+                    );
                 }
-<# } #>
             }
 
-            public void OnRemoveComponent(RemoveComponentOp op)
+            public override void OnComponentUpdate(ComponentUpdateOp op)
             {
-                Unity.Entities.Entity entity;
-                if (!view.TryGetEntity(op.EntityId.Id, out entity))
+                if (MutableView.TryGetEntity(op.EntityId, out var entity))
                 {
-                    LogDispatcher.HandleLog(LogType.Error, new LogEvent("Entity not found during OnRemoveComponent.")
+                    LogDispatcher.HandleLog(LogType.Error, new LogEvent(EntityNotFound)
                         .WithField(LoggingUtils.LoggerName, LoggerName)
                         .WithField(LoggingUtils.EntityId, op.EntityId.Id)
-                        .WithField(MutableView.Component, "<#= componentDetails.TypeName #>"));
+                        .WithField("Op Type", "ComponentUpdateOp")
+                        .WithField("Component Type", "<#= componentDetails.TypeName #>")
+                    );
                     return;
                 }
+                var data = entityManager.GetComponentData<<#= componentDetails.TypeName #>>(entity);
 
-                view.RemoveComponent<<#= componentDetails.TypeName #>>(entity);
+                var update = global::<#= qualifiedNamespace #>.<#= componentDetails.TypeName #>.Serialization.ApplyUpdate(ref data, op.Update.SchemaData.Value.GetFields());
```

Comment:
Little confusing that we get an update as a return value from a function called ApplyUdpate. Might be better to have separate functions for getting and applying updates.

Reply:
The advantage of combining them is that you don't duplicate effort - could do `GetAndApplyUpdate` instead

Reply:
I would also flip the parameter order. fields first, ref for output later

Reply:
Done 👍